File: src/main/java/me/urielsalis/cursebot/api/Channel.java
package me.urielsalis.cursebot.api;

import org.apache.commons.collections4.queue.CircularFifoQueue;

import java.util.Queue;


/**
 * CurseApi
 * @Author: Urielsalis
 * License: GPL 3.0
 */
public class Channel {
    public String groupTitle;
    public String groupID;
    public Queue<Message> messages = new CircularFifoQueue<Message>(100);
    public Channel(Object groupTitle, Object groupID) {
        this.groupID = (String) groupID;
        this.groupTitle = (String) groupTitle;
    }
}

File: src/main/java/me/urielsalis/cursebot/api/Channel.kt
package me.urielsalis.cursebot.api

public class Channel(val channelName: String, val channelUUID: String) {
    var messages: HashMap<Long, Message> = HashMap()

    override fun equals(other: Any?): Boolean {
        return other is Channel && ((other as Channel).channelUUID==this.channelUUID)
    }

    override fun hashCode(): Int {
        return channelUUID.hashCode()
    }
}